Sources/TuistKit/Services/ShareCommandService.swift

```diff
@@ -411,19 +410,30 @@ struct ShareService {
         serverURL: URL,
         json: Bool
     ) async throws {
-        ServiceContext.current?.logger?.notice("Uploading \(displayName)...")
-        let preview = try await previewsUploadService.uploadPreviews(
-            previewUploadType,
-            displayName: displayName,
-            version: version,
-            bundleIdentifier: bundleIdentifier,
-            icon: icon,
-            supportedPlatforms: supportedPlatforms,
-            fullHandle: fullHandle,
-            serverURL: serverURL
-        )
-        ServiceContext.current?.logger?
-            .notice("\(displayName) uploaded – share it with others using the following link: \(preview.url.absoluteString)")
+        let preview = try await ServiceContext.current!.ui!.progressBarStep(
+            message: "Uploading \(displayName)",
+            successMessage: "\(displayName) uploaded",
+            errorMessage: "Failed to load manifests"
+        ) { updateProgress in
+            try await previewsUploadService.uploadPreviews(
+                previewUploadType,
+                displayName: displayName,
+                version: version,
+                bundleIdentifier: bundleIdentifier,
+                icon: icon,
+                supportedPlatforms: supportedPlatforms,
+                fullHandle: fullHandle,
+                serverURL: serverURL,
+                updateProgress: updateProgress
+            )
+        }
+
+        ServiceContext.current?.ui?
+            .success(
+                .alert(
+                    "Share \(displayName) with others using the following link: \(preview.url.absoluteString)"
+                )
+            )
```

Comment:
Let's use `SerivceContext.current?.alerts?.success(...)`. It's important so that at a higher level the right order of presentation can be determined based on whether there are warnings or not.